ts/packages/node/src/stores/LaunchDarkly.ts

```diff
@@ -0,0 +1,154 @@
+      key: config.set,
+      name: config.set,
+    });
+    return config.set;
+  }
+
+  private async createFeatureFlag(config: Config) {
+    let onValue: any;
+    let offValue: any;
+    try {
+      onValue = JSON.parse(config.value);
+      if (typeof onValue === 'boolean') {
+        offValue = !onValue;
+      } else {
+        offValue = Number.isNaN(onValue) ? {} : 0;
+      }
+    } catch (e) {
+      onValue = config.value;
+      offValue = '';
+    }
+    const createData = {
+      key: config.key,
+      name: config.key,
+      variations: [{ value: onValue }, { value: offValue }],
+    };
+    const { data: featureFlag } = await this.client.post(`/flags/${this.projectKey}`, createData);
+    return { featureFlag, onValue };
+  }
+
+  private async createVariation(config: Config) {
+    let onValue: any;
+    try {
+      onValue = JSON.parse(config.value);
+    } catch (e) {
+      onValue = config.value;
+    }
+    const createVariationData = {
+      instructions: [{ kind: 'addVariation', value: onValue }],
+    };
+    const data = await this.patchUpdate(config, createVariationData);
+    return data.variations.find((variation: any) => variation.value === onValue);
+  }
+
+  private async updateDefaultFallthroughVariation(config: Config, variationId: string) {
+    const updateFallthroughVariationData = {
+      environmentKey: config.set,
+      instructions: [
+        {
+          kind: 'updateFallthroughVariationOrRollout',
+          variationId,
+        },
+      ],
+    };
+    return this.patchUpdate(config, updateFallthroughVariationData);
+  }
+
+  private async getEnvFeatureFlags(env: string, keys: string[]): Promise<Config[]> {
+    const { data: featureFlags } = await this.client.get(`/flags/${this.projectKey}?env=${env}`);
+    return featureFlags.items
+      .filter((featureFlag: any) => keys.includes(featureFlag.key))
+      .map((featureFlag: any) => {
+        const onValue = Object.entries(featureFlag.environments[env]._summary.variations)
+          .filter((value: [string, any]) => value[1].isFallthrough)
+          .map((value) => featureFlag.variations[parseInt(value[0], 10)].value)[0];
+        return { key: featureFlag.key, set: env, value: JSON.stringify(onValue) };
+      })
+      .filter((config: Config) => config.value);
+  }
+
+  async get(queries: ConfigStoreQuery[]): Promise<Config[]> {
+    if (queries.some((query) => query.set.split('/').length > 1))
+      throw new Error('Subsets are not supported in LaunchDarklyConfigStore');
+    const environments = await this.getEnvironments();
+    const noRootQueries = queries.filter((query) => query.set !== '' && environments.includes(query.set));
+    if (noRootQueries[0]) {
+      const env = noRootQueries[0].set;
+      const keys = noRootQueries.map((value) => value.key);
+      return this.getEnvFeatureFlags(env, keys);
+    }
+    throw new Error('No valid queries');
```

Comment:
When no values found. should this return empty array or something else?

Reply:
Should always return an empty array if no values are found, I also suggest we apply the following change to this method, tell me what you think:

```node.js
    const getConfigsPromises = _(queries)
      .filter((query) => query.set !== '' && environments.includes(query.set))
      .groupBy('set')
      .map((groupedConfigs) => {
        if (groupedConfigs[0]) {
          return this.getEnvFeatureFlags(groupedConfigs[0].set, _.map(groupedConfigs, 'key'));
        }
        return null;
      })
      .compact()
      .value();

    const getConfigsResults = await Promise.all(getConfigsPromises);
    const storedConfigs = _.flatten(getConfigsResults);

    return storedConfigs;
```

Reply:
Done.
Please make a note of it for future stores. `get()` should always return empty array if no exception raised.